### PR TITLE
Fix ORA-01795 while OracleSchemaMetaDataLoader.load tables count greater than 1000

### LINKED ISSUE #17250: ORA-01795 while OracleSchemaMetaDataLoader.load tables count greater than 1000
Described as title.

ENV:
5.1.2-SNAPSHOT, oracle-xe-11g-r2, ojdbc8 version 12.2.0.1

### PR DESCRIPTION
**Pull request by cheese8**
Fixes #17250.

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/loader/dialect/OracleSchemaMetaDataLoader.java [hunk 1]
```diff
@@ -66,17 +68,32 @@ public final class OracleSchemaMetaDataLoader implements DialectSchemaMetaDataLo
     
     private static final int IDENTITY_COLUMN_START_MINOR_VERSION = 1;
     
+    private static final int BATCH_SIZE = 1000;
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/loader/dialect/OracleSchemaMetaDataLoader.java:71]: @cheese8 Can we run some tests to determine which batch size perform best for metadata load?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/loader/dialect/OracleSchemaMetaDataLoader.java (hunk 1 above)
Comment (cheese8) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/loader/dialect/OracleSchemaMetaDataLoader.java:71]: The max what oracle supported is 1000 which ORA-01795 shows, and I think the less database queried the best performance get.

And `performance optimized for get connection` PRed to reuse the connection.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/loader/dialect/OracleSchemaMetaDataLoader.java [hunk 2]
```diff
@@ -66,20 +68,37 @@ public final class OracleSchemaMetaDataLoader implements DialectSchemaMetaDataLo
...
+    
     @Override
     public Collection<SchemaMetaData> load(final DataSource dataSource, final Collection<String> tables, final String defaultSchemaName) throws SQLException {
         Map<String, TableMetaData> tableMetaDataMap = new LinkedHashMap<>();
-        Map<String, Collection<ColumnMetaData>> columnMetaDataMap = loadColumnMetaDataMap(dataSource, tables);
+        Map<String, Collection<ColumnMetaData>> columnMetaDataMap = new HashMap<>(tables.size());
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/loader/dialect/OracleSchemaMetaDataLoader.java:76]: @cheese8 Please add load factor when using new HashMap().

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/loader/dialect/OracleSchemaMetaDataLoader.java [hunk 3]
```diff
@@ -66,20 +68,37 @@ public final class OracleSchemaMetaDataLoader implements DialectSchemaMetaDataLo
...
+        Collection[] splitTables = splitTables(new ArrayList(tables), BATCH_SIZE);
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/loader/dialect/OracleSchemaMetaDataLoader.java:77]: @cheese8 Can we use guava `Lists.partition` to replace this method?

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/loader/dialect/OracleSchemaMetaDataLoader.java [hunk 4]
```diff
@@ -66,20 +68,37 @@ public final class OracleSchemaMetaDataLoader implements DialectSchemaMetaDataLo
...
+        try (Connection connection = dataSource.getConnection()) {
+            for (Collection<String> subTables : splitTables) {
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/loader/dialect/OracleSchemaMetaDataLoader.java:79]: Please rename subTables to each.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/loader/dialect/OracleSchemaMetaDataLoader.java [hunk 5]
```diff
@@ -66,20 +69,28 @@ public final class OracleSchemaMetaDataLoader implements DialectSchemaMetaDataLo
     
     private static final int IDENTITY_COLUMN_START_MINOR_VERSION = 1;
     
+    private static final int BATCH_SIZE = 1000;
+    
     @Override
     public Collection<SchemaMetaData> load(final DataSource dataSource, final Collection<String> tables, final String defaultSchemaName) throws SQLException {
         Map<String, TableMetaData> tableMetaDataMap = new LinkedHashMap<>();
-        Map<String, Collection<ColumnMetaData>> columnMetaDataMap = loadColumnMetaDataMap(dataSource, tables);
+        Map<String, Collection<ColumnMetaData>> columnMetaDataMap = new HashMap<>(tables.size(), 0.75f);
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/loader/dialect/OracleSchemaMetaDataLoader.java:77]: Why set load factor to 0.75? Not 1?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/loader/dialect/OracleSchemaMetaDataLoader.java (hunk 5 above)
Comment (cheese8) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/schema/loader/dialect/OracleSchemaMetaDataLoader.java]: Not thinking of it, default 0.75 used. Fixed.

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**